// to-excel/lib/cjs/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_VERSION = exports.DEFAULT_COMPANY = exports.DEFAULT_AUTHOR = exports.X_ROW_END = exports.X_ROW_START = exports.DEFAULT_EXT = exports.X_CELL_END = exports.X_CELL_START = exports.X_FOOTER = exports.X_STYLES = exports.X_PROPS = exports.X_HEADER = void 0;
exports.X_HEADER = `<?xml version="1.0"?>\n<?mso-application progid="Excel.Sheet"?>
<Workbook\nxmlns="urn:schemas-microsoft-com:office:spreadsheet"\nxmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"\nxmlns:ss="urn:schemas-microsoft-com:office:spreadsheet"
xmlns:html="http://www.w3.org/TR/REC-html40">\n`;
exports.X_PROPS = `<DocumentProperties xmlns="urn:schemas-microsoft-com:office:office">
<Author>%author</Author>
<LastAuthor>%lastAuthor</LastAuthor>
<Company>%company</Company>
<Version>%version</Version>
</DocumentProperties>\n`;
exports.X_STYLES = `<Styles>
<Style ss:ID="hdr"><Font ss:Color="#ffffff" ss:Bold="1" /><Interior ss:Color="#000000" ss:Pattern="Solid"/></Style>
</Styles>`;
exports.X_FOOTER = '\n</Table></Worksheet>\n</Workbook>';
exports.X_CELL_START = '\n<Cell ss:StyleID="hdr"><Data ss:Type="String">';
exports.X_CELL_END = '</Data></Cell>';
exports.DEFAULT_EXT = 'xls';
exports.X_ROW_START = '\n<Row>';
exports.X_ROW_END = '\n</Row>';
exports.DEFAULT_AUTHOR = 'Ricard Fíguls';
exports.DEFAULT_COMPANY = 'RFM Software (Ricard Figuls)';
exports.DEFAULT_VERSION = '1';
