// @ironsoftware/ironpdf/src/public/naturalLanguages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NaturalLanguages = void 0;
/**
 * Natural Languages Specification
 */
var NaturalLanguages;
(function (NaturalLanguages) {
    //#region Non-specific Region
    /**
     * Arabic Language
     */
    NaturalLanguages[NaturalLanguages["Arabic"] = 1] = "Arabic";
    /**
     * Bengali Language
     */
    NaturalLanguages[NaturalLanguages["Bengali"] = 2] = "Bengali";
    /**
     * Chamorro Language
     */
    NaturalLanguages[NaturalLanguages["Chamorro"] = 3] = "Chamorro";
    /**
     * Danish Language
     */
    NaturalLanguages[NaturalLanguages["Danish"] = 4] = "Danish";
    /**
     * German Language
     */
    NaturalLanguages[NaturalLanguages["German"] = 5] = "German";
    /**
     * Greek Language
     */
    NaturalLanguages[NaturalLanguages["Greek"] = 6] = "Greek";
    /**
     * English Language
     */
    NaturalLanguages[NaturalLanguages["English"] = 7] = "English";
    /**
     * Spanish Language
     */
    NaturalLanguages[NaturalLanguages["Spanish"] = 8] = "Spanish";
    /**
     * French Language
     */
    NaturalLanguages[NaturalLanguages["French"] = 9] = "French";
    /**
     * Irish Language
     */
    NaturalLanguages[NaturalLanguages["Irish"] = 10] = "Irish";
    /**
     * Croatian Language
     */
    NaturalLanguages[NaturalLanguages["Croatian"] = 11] = "Croatian";
    /**
     * Hungarian Language
     */
    NaturalLanguages[NaturalLanguages["Hungarian"] = 12] = "Hungarian";
    /**
     * Italian Language
     */
    NaturalLanguages[NaturalLanguages["Italian"] = 13] = "Italian";
    /**
     * Korean Language
     */
    NaturalLanguages[NaturalLanguages["Korean"] = 14] = "Korean";
    /**
     * Lingala Language
     */
    NaturalLanguages[NaturalLanguages["Lingala"] = 15] = "Lingala";
    /**
     * Malay Language
     */
    NaturalLanguages[NaturalLanguages["Malay"] = 16] = "Malay";
    /**
     * Dutch Language
     */
    NaturalLanguages[NaturalLanguages["Dutch"] = 17] = "Dutch";
    /**
     * Portuguese Language
     */
    NaturalLanguages[NaturalLanguages["Portuguese"] = 18] = "Portuguese";
    /**
     * Sindhi Language
     */
    NaturalLanguages[NaturalLanguages["Sindhi"] = 19] = "Sindhi";
    /**
     * Serbian Language
     */
    NaturalLanguages[NaturalLanguages["Serbian"] = 20] = "Serbian";
    /**
     * Swati Language
     */
    NaturalLanguages[NaturalLanguages["Swati"] = 21] = "Swati";
    /**
     * Swedish Language
     */
    NaturalLanguages[NaturalLanguages["Swedish"] = 22] = "Swedish";
    /**
     * Swahili Language
     */
    NaturalLanguages[NaturalLanguages["Swahili"] = 23] = "Swahili";
    /**
     * Tamil Language
     */
    NaturalLanguages[NaturalLanguages["Tamil"] = 24] = "Tamil";
    /**
     * Tswana Language
     */
    NaturalLanguages[NaturalLanguages["Tswana"] = 25] = "Tswana";
    /**
     * Turkish Language
     */
    NaturalLanguages[NaturalLanguages["Turkish"] = 26] = "Turkish";
    /**
     * Urdu Language
     */
    NaturalLanguages[NaturalLanguages["Urdu"] = 27] = "Urdu";
    /**
     * Mandarin Chinese Language
     */
    NaturalLanguages[NaturalLanguages["MandarinChinese"] = 28] = "MandarinChinese";
    //#endregion
    //#region Specific Region
    //#region Arabic
    /**
     * Arabic Language for United Arab Emirates Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_UnitedArabEmirates"] = 29] = "Arabic_UnitedArabEmirates";
    /**
     * Arabic Language for Bahrain Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Bahrain"] = 30] = "Arabic_Bahrain";
    /**
     * Arabic Language for Algeria Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Algeria"] = 31] = "Arabic_Algeria";
    /**
     * Arabic Language for Egypt Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Egypt"] = 32] = "Arabic_Egypt";
    /**
     * Arabic Language for Israel Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Israel"] = 33] = "Arabic_Israel";
    /**
     * Arabic Language for Iraq Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Iraq"] = 34] = "Arabic_Iraq";
    /**
     * Arabic Language for Jordan Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Jordan"] = 35] = "Arabic_Jordan";
    /**
     * Arabic Language for Kuwait Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Kuwait"] = 36] = "Arabic_Kuwait";
    /**
     * Arabic Language for Lebanon Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Lebanon"] = 37] = "Arabic_Lebanon";
    /**
     * Arabic Language for Libya Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Libya"] = 38] = "Arabic_Libya";
    /**
     * Arabic Language for Morocco Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Morocco"] = 39] = "Arabic_Morocco";
    /**
     * Arabic Language for Mauritania Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Mauritania"] = 40] = "Arabic_Mauritania";
    /**
     * Arabic Language for Oman Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Oman"] = 41] = "Arabic_Oman";
    /**
     * Arabic Language for Palestinian West Bank and Gaza Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_PalestinianWestBankAndGaza"] = 42] = "Arabic_PalestinianWestBankAndGaza";
    /**
     * Arabic Language for Qatar Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Qatar"] = 43] = "Arabic_Qatar";
    /**
     * Arabic Language for Saudi Arabia Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_SaudiArabia"] = 44] = "Arabic_SaudiArabia";
    /**
     * Arabic Language for Sudan Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Sudan"] = 45] = "Arabic_Sudan";
    /**
     * Arabic Language for Somalia Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Somalia"] = 46] = "Arabic_Somalia";
    /**
     * Arabic Language for Syria Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Syria"] = 47] = "Arabic_Syria";
    /**
     * Arabic Language for Chad Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Chad"] = 48] = "Arabic_Chad";
    /**
     * Arabic Language for Tunisia Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Tunisia"] = 49] = "Arabic_Tunisia";
    /**
     * Arabic Language for Yemen Region
     */
    NaturalLanguages[NaturalLanguages["Arabic_Yemen"] = 50] = "Arabic_Yemen";
    //#endregion
    //#region Bengali
    /**
     * Bengali Language for Bangladesh Region
     */
    NaturalLanguages[NaturalLanguages["Bengali_Bangladesh"] = 51] = "Bengali_Bangladesh";
    /**
     * Bengali Language for India Region
     */
    NaturalLanguages[NaturalLanguages["Bengali_India"] = 52] = "Bengali_India";
    /**
     * Bengali Language for Singapore Region
     */
    NaturalLanguages[NaturalLanguages["Bengali_Singapore"] = 53] = "Bengali_Singapore";
    //#endregion
    //#region Chamorro
    /**
     * Chamorro Language for Guam Region
     */
    NaturalLanguages[NaturalLanguages["Chamorro_Guam"] = 54] = "Chamorro_Guam";
    /**
     * Chamorro Language for Northern Mariana Islands Region
     */
    NaturalLanguages[NaturalLanguages["Chamorro_NorthernMarianaIslands"] = 55] = "Chamorro_NorthernMarianaIslands";
    //#endregion
    //#region Welsh
    /**
     * Welsh Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["Welsh_UnitedKingdom"] = 56] = "Welsh_UnitedKingdom";
    //#endregion
    //#region Danish
    /**
     * Danish Language for Denmark Region
     */
    NaturalLanguages[NaturalLanguages["Danish_Denmark"] = 57] = "Danish_Denmark";
    /**
     * Danish Language for Greenland Region
     */
    NaturalLanguages[NaturalLanguages["Danish_Greenland"] = 58] = "Danish_Greenland";
    //#endregion
    //#region German
    /**
     * German Language for Austria Region
     */
    NaturalLanguages[NaturalLanguages["German_Austria"] = 59] = "German_Austria";
    /**
     * German Language for Belgium Region
     */
    NaturalLanguages[NaturalLanguages["German_Belgium"] = 60] = "German_Belgium";
    /**
     * German Language for Switzerland Region
     */
    NaturalLanguages[NaturalLanguages["German_Switzerland"] = 61] = "German_Switzerland";
    /**
     * German Language for Germany Region
     */
    NaturalLanguages[NaturalLanguages["German_Germany"] = 62] = "German_Germany";
    /**
     * German Language for Denmark Region
     */
    NaturalLanguages[NaturalLanguages["German_Denmark"] = 63] = "German_Denmark";
    /**
     * German Language for Liechtensein Region
     */
    NaturalLanguages[NaturalLanguages["German_Liechtensein"] = 64] = "German_Liechtensein";
    /**
     * German Language for Luxembourg Region
     */
    NaturalLanguages[NaturalLanguages["German_Luxembourg"] = 65] = "German_Luxembourg";
    //#endregion
    //#region Greek
    /**
     * Greek Language for Cyprus Region
     */
    NaturalLanguages[NaturalLanguages["Greek_Cyprus"] = 66] = "Greek_Cyprus";
    /**
     * Greek Language for Greece Region
     */
    NaturalLanguages[NaturalLanguages["Greek_Greece"] = 67] = "Greek_Greece";
    //#endregion
    //#region English
    /**
     * English Language for Antigua and Barbuda Region
     */
    NaturalLanguages[NaturalLanguages["English_AntiguaAndBarbuda"] = 68] = "English_AntiguaAndBarbuda";
    /**
     * English Language for Anguilla Region
     */
    NaturalLanguages[NaturalLanguages["English_Anguilla"] = 69] = "English_Anguilla";
    /**
     * English Language for American Samoa Region
     */
    NaturalLanguages[NaturalLanguages["English_AmericanSamoa"] = 70] = "English_AmericanSamoa";
    /**
     * English Language for Australia Region
     */
    NaturalLanguages[NaturalLanguages["English_Australia"] = 71] = "English_Australia";
    /**
     * English Language for Barbados Region
     */
    NaturalLanguages[NaturalLanguages["English_Barbados"] = 72] = "English_Barbados";
    /**
     * English Language for Bermuda Region
     */
    NaturalLanguages[NaturalLanguages["English_Bermuda"] = 73] = "English_Bermuda";
    /**
     * English Language for Brunei Region
     */
    NaturalLanguages[NaturalLanguages["English_Brunei"] = 74] = "English_Brunei";
    /**
     * English Language for Bahamas Region
     */
    NaturalLanguages[NaturalLanguages["English_Bahamas"] = 75] = "English_Bahamas";
    /**
     * English Language for Botswana Region
     */
    NaturalLanguages[NaturalLanguages["English_Botswana"] = 76] = "English_Botswana";
    /**
     * English Language for Belize Region
     */
    NaturalLanguages[NaturalLanguages["English_Belize"] = 77] = "English_Belize";
    /**
     * English Language for Canada Region
     */
    NaturalLanguages[NaturalLanguages["English_Canada"] = 78] = "English_Canada";
    /**
     * English Language for Cook Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_CookIslands"] = 79] = "English_CookIslands";
    /**
     * English Language for Cameroon Region
     */
    NaturalLanguages[NaturalLanguages["English_Cameroon"] = 80] = "English_Cameroon";
    /**
     * English Language for Dominica Region
     */
    NaturalLanguages[NaturalLanguages["English_Dominica"] = 81] = "English_Dominica";
    /**
     * English Language for Eritrea Region
     */
    NaturalLanguages[NaturalLanguages["English_Eritrea"] = 82] = "English_Eritrea";
    /**
     * English Language for Ethiopia Region
     */
    NaturalLanguages[NaturalLanguages["English_Ethiopia"] = 83] = "English_Ethiopia";
    /**
     * English Language for Fiji Region
     */
    NaturalLanguages[NaturalLanguages["English_Fiji"] = 84] = "English_Fiji";
    /**
     * English Language for Falkland Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_FalklandIslands"] = 85] = "English_FalklandIslands";
    /**
     * English Language for Micronesia Region
     */
    NaturalLanguages[NaturalLanguages["English_Micronesia"] = 86] = "English_Micronesia";
    /**
     * English Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["English_UnitedKingdom"] = 87] = "English_UnitedKingdom";
    /**
     * English Language for Grenada Region
     */
    NaturalLanguages[NaturalLanguages["English_Grenada"] = 88] = "English_Grenada";
    /**
     * English Language for Ghana Region
     */
    NaturalLanguages[NaturalLanguages["English_Ghana"] = 89] = "English_Ghana";
    /**
     * English Language for Gibraltar Region
     */
    NaturalLanguages[NaturalLanguages["English_Gibraltar"] = 90] = "English_Gibraltar";
    /**
     * English Language for Gambia Region
     */
    NaturalLanguages[NaturalLanguages["English_Gambia"] = 91] = "English_Gambia";
    /**
     * English Language for Guam Region
     */
    NaturalLanguages[NaturalLanguages["English_Guam"] = 92] = "English_Guam";
    /**
     * English Language for Guyana Region
     */
    NaturalLanguages[NaturalLanguages["English_Guyana"] = 93] = "English_Guyana";
    /**
     * English Language for Ireland Region
     */
    NaturalLanguages[NaturalLanguages["English_Ireland"] = 94] = "English_Ireland";
    /**
     * English Language for Israel Region
     */
    NaturalLanguages[NaturalLanguages["English_Israel"] = 95] = "English_Israel";
    /**
     * English Language for British Indian Ocean Territ Region
     */
    NaturalLanguages[NaturalLanguages["English_BritishIndianOceanTerrit"] = 96] = "English_BritishIndianOceanTerrit";
    /**
     * English Language for Jamaica Region
     */
    NaturalLanguages[NaturalLanguages["English_Jamaica"] = 97] = "English_Jamaica";
    /**
     * English Language for Kenya Region
     */
    NaturalLanguages[NaturalLanguages["English_Kenya"] = 98] = "English_Kenya";
    /**
     * English Language for Kiribati Region
     */
    NaturalLanguages[NaturalLanguages["English_Kiribati"] = 99] = "English_Kiribati";
    /**
     * English Language for St. Kitts-Nevis Region
     */
    NaturalLanguages[NaturalLanguages["English_StKittsNevis"] = 100] = "English_StKittsNevis";
    /**
     * English Language for Cayman Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_CaymanIslands"] = 101] = "English_CaymanIslands";
    /**
     * English Language for St. Lucia Region
     */
    NaturalLanguages[NaturalLanguages["English_StLucia"] = 102] = "English_StLucia";
    /**
     * English Language for Liberia Region
     */
    NaturalLanguages[NaturalLanguages["English_Liberia"] = 103] = "English_Liberia";
    /**
     * English Language for Lesotho Region
     */
    NaturalLanguages[NaturalLanguages["English_Lesotho"] = 104] = "English_Lesotho";
    /**
     * English Language for Northern Mariana Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_NorthernMarianaIslands"] = 105] = "English_NorthernMarianaIslands";
    /**
     * English Language for Montserrat Region
     */
    NaturalLanguages[NaturalLanguages["English_Montserrat"] = 106] = "English_Montserrat";
    /**
     * English Language for Malta Region
     */
    NaturalLanguages[NaturalLanguages["English_Malta"] = 107] = "English_Malta";
    /**
     * English Language for Mauritius Region
     */
    NaturalLanguages[NaturalLanguages["English_Mauritius"] = 108] = "English_Mauritius";
    /**
     * English Language for Malawi Region
     */
    NaturalLanguages[NaturalLanguages["English_Malawi"] = 109] = "English_Malawi";
    /**
     * English Language for Namibia Region
     */
    NaturalLanguages[NaturalLanguages["English_Namibia"] = 110] = "English_Namibia";
    /**
     * English Language for Norfolk Island Region
     */
    NaturalLanguages[NaturalLanguages["English_NorfolkIsland"] = 111] = "English_NorfolkIsland";
    /**
     * English Language for Nigeria Region
     */
    NaturalLanguages[NaturalLanguages["English_Nigeria"] = 112] = "English_Nigeria";
    /**
     * English Language for Nauru Region
     */
    NaturalLanguages[NaturalLanguages["English_Nauru"] = 113] = "English_Nauru";
    /**
     * English Language for Niue Region
     */
    NaturalLanguages[NaturalLanguages["English_Niue"] = 114] = "English_Niue";
    /**
     * English Language for New Zealand Region
     */
    NaturalLanguages[NaturalLanguages["English_NewZealand"] = 115] = "English_NewZealand";
    /**
     * English Language for Papua New Guinea Region
     */
    NaturalLanguages[NaturalLanguages["English_PapuaNewGuinea"] = 116] = "English_PapuaNewGuinea";
    /**
     * English Language for Philippines Region
     */
    NaturalLanguages[NaturalLanguages["English_Philippines"] = 117] = "English_Philippines";
    /**
     * English Language for Pakistan Region
     */
    NaturalLanguages[NaturalLanguages["English_Pakistan"] = 118] = "English_Pakistan";
    /**
     * English Language for Pitcairn Region
     */
    NaturalLanguages[NaturalLanguages["English_Pitcairn"] = 119] = "English_Pitcairn";
    /**
     * English Language for Puerto Rico Region
     */
    NaturalLanguages[NaturalLanguages["English_PuertoRico"] = 120] = "English_PuertoRico";
    /**
     * English Language for Palau Region
     */
    NaturalLanguages[NaturalLanguages["English_Palau"] = 121] = "English_Palau";
    /**
     * English Language for Rwanda Region
     */
    NaturalLanguages[NaturalLanguages["English_Rwanda"] = 122] = "English_Rwanda";
    /**
     * English Language for Solomon Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_SolomonIslands"] = 123] = "English_SolomonIslands";
    /**
     * English Language for Seychelles Region
     */
    NaturalLanguages[NaturalLanguages["English_Seychelles"] = 124] = "English_Seychelles";
    /**
     * English Language for Singapore Region
     */
    NaturalLanguages[NaturalLanguages["English_Singapore"] = 125] = "English_Singapore";
    /**
     * English Language for St. Helena Region
     */
    NaturalLanguages[NaturalLanguages["English_StHelena"] = 126] = "English_StHelena";
    /**
     * English Language for Sierra Leone Region
     */
    NaturalLanguages[NaturalLanguages["English_SierraLeone"] = 127] = "English_SierraLeone";
    /**
     * English Language for Somalia Region
     */
    NaturalLanguages[NaturalLanguages["English_Somalia"] = 128] = "English_Somalia";
    /**
     * English Language for Swaziland Region
     */
    NaturalLanguages[NaturalLanguages["English_Swaziland"] = 129] = "English_Swaziland";
    /**
     * English Language for Turks and Caicos Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_TurksAndCaicosIslands"] = 130] = "English_TurksAndCaicosIslands";
    /**
     * English Language for Tokelau Region
     */
    NaturalLanguages[NaturalLanguages["English_Tokelau"] = 131] = "English_Tokelau";
    /**
     * English Language for Tonga Region
     */
    NaturalLanguages[NaturalLanguages["English_Tonga"] = 132] = "English_Tonga";
    /**
     * English Language for Trinidad and Tobago Region
     */
    NaturalLanguages[NaturalLanguages["English_TrinidadAndTobago"] = 133] = "English_TrinidadAndTobago";
    /**
     * English Language for Uganda Region
     */
    NaturalLanguages[NaturalLanguages["English_Uganda"] = 134] = "English_Uganda";
    /**
     * English Language for United States of America Region
     */
    NaturalLanguages[NaturalLanguages["English_USA"] = 135] = "English_USA";
    /**
     * English Language for St. Vincent and the Grenadi Region
     */
    NaturalLanguages[NaturalLanguages["English_StVincentAndTheGrenadi"] = 136] = "English_StVincentAndTheGrenadi";
    /**
     * English Language for British Virgin Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_BritishVirginIslands"] = 137] = "English_BritishVirginIslands";
    /**
     * English Language for U.S. Virgin Islands Region
     */
    NaturalLanguages[NaturalLanguages["English_USVirginIslands"] = 138] = "English_USVirginIslands";
    /**
     * English Language for Vanuatu Region
     */
    NaturalLanguages[NaturalLanguages["English_Vanuatu"] = 139] = "English_Vanuatu";
    /**
     * English Language for Western Samoa Region
     */
    NaturalLanguages[NaturalLanguages["English_WesternSamoa"] = 140] = "English_WesternSamoa";
    /**
     * English Language for South Africa Region
     */
    NaturalLanguages[NaturalLanguages["English_SouthAfrica"] = 141] = "English_SouthAfrica";
    /**
     * English Language for Zambia Region
     */
    NaturalLanguages[NaturalLanguages["English_Zambia"] = 142] = "English_Zambia";
    /**
     * English Language for Zimbabwe Region
     */
    NaturalLanguages[NaturalLanguages["English_Zimbabwe"] = 143] = "English_Zimbabwe";
    //#endregion
    //#region Spanish
    /**
     * Spanish Language for Argentina Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Argentina"] = 144] = "Spanish_Argentina";
    /**
     * Spanish Language for Bolivia Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Bolivia"] = 145] = "Spanish_Bolivia";
    /**
     * Spanish Language for Chile Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Chile"] = 146] = "Spanish_Chile";
    /**
     * Spanish Language for Colombia Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Colombia"] = 147] = "Spanish_Colombia";
    /**
     * Spanish Language for Costa Rica Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_CostaRica"] = 148] = "Spanish_CostaRica";
    /**
     * Spanish Language for Cuba Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Cuba"] = 149] = "Spanish_Cuba";
    /**
     * Spanish Language for Dominican Republic Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_DominicanRepublic"] = 150] = "Spanish_DominicanRepublic";
    /**
     * Spanish Language for Ecuador Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Ecuador"] = 151] = "Spanish_Ecuador";
    /**
     * Spanish Language for Spain Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Spain"] = 152] = "Spanish_Spain";
    /**
     * Spanish Language for Equatorial Guinea Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_EquatorialGuinea"] = 153] = "Spanish_EquatorialGuinea";
    /**
     * Spanish Language for Guatemala Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Guatemala"] = 154] = "Spanish_Guatemala";
    /**
     * Spanish Language for Honduras Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Honduras"] = 155] = "Spanish_Honduras";
    /**
     * Spanish Language for Mexico Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Mexico"] = 156] = "Spanish_Mexico";
    /**
     * Spanish Language for Nicaragua Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Nicaragua"] = 157] = "Spanish_Nicaragua";
    /**
     * Spanish Language for Panama Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Panama"] = 158] = "Spanish_Panama";
    /**
     * Spanish Language for Peru Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Peru"] = 159] = "Spanish_Peru";
    /**
     * Spanish Language for Puerto Rico Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_PuertoRico"] = 160] = "Spanish_PuertoRico";
    /**
     * Spanish Language for Paraguay Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Paraguay"] = 161] = "Spanish_Paraguay";
    /**
     * Spanish Language for El Salvador Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_ElSalvador"] = 162] = "Spanish_ElSalvador";
    /**
     * Spanish Language for United States of America Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_USA"] = 163] = "Spanish_USA";
    /**
     * Spanish Language for Uruguay Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Uruguay"] = 164] = "Spanish_Uruguay";
    /**
     * Spanish Language for Venezuela Region
     */
    NaturalLanguages[NaturalLanguages["Spanish_Venezuela"] = 165] = "Spanish_Venezuela";
    //#endregion
    //#region French
    /**
     * French Language for Andorra Region
     */
    NaturalLanguages[NaturalLanguages["French_Andorra"] = 166] = "French_Andorra";
    /**
     * French Language for Belgium Region
     */
    NaturalLanguages[NaturalLanguages["French_Belgium"] = 167] = "French_Belgium";
    /**
     * French Language for Burkina Faso Region
     */
    NaturalLanguages[NaturalLanguages["French_BurkinaFaso"] = 168] = "French_BurkinaFaso";
    /**
     * French Language for Burundi Region
     */
    NaturalLanguages[NaturalLanguages["French_Burundi"] = 169] = "French_Burundi";
    /**
     * French Language for Benin Region
     */
    NaturalLanguages[NaturalLanguages["French_Benin"] = 170] = "French_Benin";
    /**
     * French Language for Canada Region
     */
    NaturalLanguages[NaturalLanguages["French_Canada"] = 171] = "French_Canada";
    /**
     * French Language for Democratic Republic of Congo Region
     */
    NaturalLanguages[NaturalLanguages["French_DemocraticRepublicOfCongo"] = 172] = "French_DemocraticRepublicOfCongo";
    /**
     * French Language for Central African Republic Region
     */
    NaturalLanguages[NaturalLanguages["French_CentralAfricanRepublic"] = 173] = "French_CentralAfricanRepublic";
    /**
     * French Language for Congo Region
     */
    NaturalLanguages[NaturalLanguages["French_Congo"] = 174] = "French_Congo";
    /**
     * French Language for Switzerland Region
     */
    NaturalLanguages[NaturalLanguages["French_Switzerland"] = 175] = "French_Switzerland";
    /**
     * French Language for Cote d'Ivoire Region
     */
    NaturalLanguages[NaturalLanguages["French_CoteDIvoire"] = 176] = "French_CoteDIvoire";
    /**
     * French Language for Cameroon Region
     */
    NaturalLanguages[NaturalLanguages["French_Cameroon"] = 177] = "French_Cameroon";
    /**
     * French Language for Djibouti Region
     */
    NaturalLanguages[NaturalLanguages["French_Djibouti"] = 178] = "French_Djibouti";
    /**
     * French Language for France Region
     */
    NaturalLanguages[NaturalLanguages["French_France"] = 179] = "French_France";
    /**
     * French Language for Gabon Region
     */
    NaturalLanguages[NaturalLanguages["French_Gabon"] = 180] = "French_Gabon";
    /**
     * French Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["French_UnitedKingdom"] = 181] = "French_UnitedKingdom";
    /**
     * French Language for French Guiana Region
     */
    NaturalLanguages[NaturalLanguages["French_FrenchGuiana"] = 182] = "French_FrenchGuiana";
    /**
     * French Language for Guinea Region
     */
    NaturalLanguages[NaturalLanguages["French_Guinea"] = 183] = "French_Guinea";
    /**
     * French Language for Guadeloupe Region
     */
    NaturalLanguages[NaturalLanguages["French_Guadeloupe"] = 184] = "French_Guadeloupe";
    /**
     * French Language for Haiti Region
     */
    NaturalLanguages[NaturalLanguages["French_Haiti"] = 185] = "French_Haiti";
    /**
     * French Language for Italy Region
     */
    NaturalLanguages[NaturalLanguages["French_Italy"] = 186] = "French_Italy";
    /**
     * French Language for Comoros Islands Region
     */
    NaturalLanguages[NaturalLanguages["French_ComorosIslands"] = 187] = "French_ComorosIslands";
    /**
     * French Language for Lebanon Region
     */
    NaturalLanguages[NaturalLanguages["French_Lebanon"] = 188] = "French_Lebanon";
    /**
     * French Language for Luxembourg Region
     */
    NaturalLanguages[NaturalLanguages["French_Luxembourg"] = 189] = "French_Luxembourg";
    /**
     * French Language for Monaco Region
     */
    NaturalLanguages[NaturalLanguages["French_Monaco"] = 190] = "French_Monaco";
    /**
     * French Language for Madagascar Region
     */
    NaturalLanguages[NaturalLanguages["French_Madagascar"] = 191] = "French_Madagascar";
    /**
     * French Language for Mali Region
     */
    NaturalLanguages[NaturalLanguages["French_Mali"] = 192] = "French_Mali";
    /**
     * French Language for Martinique Region
     */
    NaturalLanguages[NaturalLanguages["French_Martinique"] = 193] = "French_Martinique";
    /**
     * French Language for New Caledonia Region
     */
    NaturalLanguages[NaturalLanguages["French_NewCaledonia"] = 194] = "French_NewCaledonia";
    /**
     * French Language for French Polynesia Region
     */
    NaturalLanguages[NaturalLanguages["French_FrenchPolynesia"] = 195] = "French_FrenchPolynesia";
    /**
     * French Language for St. Pierre and Miquelon Region
     */
    NaturalLanguages[NaturalLanguages["French_StPierreAndMiquelon"] = 196] = "French_StPierreAndMiquelon";
    /**
     * French Language for Reunion Region
     */
    NaturalLanguages[NaturalLanguages["French_Reunion"] = 197] = "French_Reunion";
    /**
     * French Language for Rwanda Region
     */
    NaturalLanguages[NaturalLanguages["French_Rwanda"] = 198] = "French_Rwanda";
    /**
     * French Language for Seychelles Region
     */
    NaturalLanguages[NaturalLanguages["French_Seychelles"] = 199] = "French_Seychelles";
    /**
     * French Language for Chad Region
     */
    NaturalLanguages[NaturalLanguages["French_Chad"] = 200] = "French_Chad";
    /**
     * French Language for Togo Region
     */
    NaturalLanguages[NaturalLanguages["French_Togo"] = 201] = "French_Togo";
    /**
     * French Language for Vanuatu Region
     */
    NaturalLanguages[NaturalLanguages["French_Vanuatu"] = 202] = "French_Vanuatu";
    /**
     * French Language for Wallis and Futuna Region
     */
    NaturalLanguages[NaturalLanguages["French_WallisAndFutuna"] = 203] = "French_WallisAndFutuna";
    /**
     * French Language for Mayotte Region
     */
    NaturalLanguages[NaturalLanguages["French_Mayotte"] = 204] = "French_Mayotte";
    //#endregion
    //#region Irish
    /**
     * Irish Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["Irish_UnitedKingdom"] = 205] = "Irish_UnitedKingdom";
    /**
     * Irish Language for Ireland Region
     */
    NaturalLanguages[NaturalLanguages["Irish_Ireland"] = 206] = "Irish_Ireland";
    //#endregion
    //#region Gaelic
    /**
     * Gaelic Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["Gaelic_UnitedKingdom"] = 207] = "Gaelic_UnitedKingdom";
    //#endregion
    //#region Manx
    /**
     * Manx Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["Manx_UnitedKingdom"] = 208] = "Manx_UnitedKingdom";
    //#endregion
    //#region Croatian
    /**
     * Croatian Language for Bosnia-Herzegovina Region
     */
    NaturalLanguages[NaturalLanguages["Croatian_BosniaHerzegovina"] = 209] = "Croatian_BosniaHerzegovina";
    /**
     * Croatian Language for Croatia Region
     */
    NaturalLanguages[NaturalLanguages["Croatian_Croatia"] = 210] = "Croatian_Croatia";
    //#endregion
    //#region Hungarian
    /**
     * Hungarian Language for Hungary Region
     */
    NaturalLanguages[NaturalLanguages["Hungarian_Hungary"] = 211] = "Hungarian_Hungary";
    /**
     * Hungarian Language for Slovenia Region
     */
    NaturalLanguages[NaturalLanguages["Hungarian_Slovenia"] = 212] = "Hungarian_Slovenia";
    //#endregion
    //#region Italian
    /**
     * Italian Language for Switzerland Region
     */
    NaturalLanguages[NaturalLanguages["Italian_Switzerland"] = 213] = "Italian_Switzerland";
    /**
     * Italian Language for Croatia Region
     */
    NaturalLanguages[NaturalLanguages["Italian_Croatia"] = 214] = "Italian_Croatia";
    /**
     * Italian Language for Italy Region
     */
    NaturalLanguages[NaturalLanguages["Italian_Italy"] = 215] = "Italian_Italy";
    /**
     * Italian Language for Slovenia Region
     */
    NaturalLanguages[NaturalLanguages["Italian_Slovenia"] = 216] = "Italian_Slovenia";
    /**
     * Italian Language for San Marino Region
     */
    NaturalLanguages[NaturalLanguages["Italian_SanMarino"] = 217] = "Italian_SanMarino";
    //#endregion
    //#region Korean
    /**
     * Korean Language for Korea, North Region
     */
    NaturalLanguages[NaturalLanguages["Korean_KoreaNorth"] = 218] = "Korean_KoreaNorth";
    /**
     * Korean Language for Korea, South Region
     */
    NaturalLanguages[NaturalLanguages["Korean_KoreaSouth"] = 219] = "Korean_KoreaSouth";
    //#endregion
    //#region Cornish
    /**
     * Cornish Language for United Kingdom Region
     */
    NaturalLanguages[NaturalLanguages["Cornish_UnitedKingdom"] = 220] = "Cornish_UnitedKingdom";
    //#endregion
    //#region Lingala
    /**
     * Lingala Language for Democratic Republic of Congo Region
     */
    NaturalLanguages[NaturalLanguages["Lingala_DemocraticRepublicOfCongo"] = 221] = "Lingala_DemocraticRepublicOfCongo";
    /**
     * Lingala Language for Congo Region
     */
    NaturalLanguages[NaturalLanguages["Lingala_Congo"] = 222] = "Lingala_Congo";
    //#endregion
    //#region Malay
    /**
     * Malay Language for Brunei Region
     */
    NaturalLanguages[NaturalLanguages["Malay_Brunei"] = 223] = "Malay_Brunei";
    /**
     * Malay Language for Malaysia Region
     */
    NaturalLanguages[NaturalLanguages["Malay_Malaysia"] = 224] = "Malay_Malaysia";
    /**
     * Malay Language for Singapore Region
     */
    NaturalLanguages[NaturalLanguages["Malay_Singapore"] = 225] = "Malay_Singapore";
    //#endregion
    //#region Dutch
    /**
     * Dutch Language for Netherlands Antilles Region
     */
    NaturalLanguages[NaturalLanguages["Dutch_NetherlandsAntilles"] = 226] = "Dutch_NetherlandsAntilles";
    /**
     * Dutch Language for Aruba Region
     */
    NaturalLanguages[NaturalLanguages["Dutch_Aruba"] = 227] = "Dutch_Aruba";
    /**
     * Dutch Language for Belgium Region
     */
    NaturalLanguages[NaturalLanguages["Dutch_Belgium"] = 228] = "Dutch_Belgium";
    /**
     * Dutch Language for Netherlands Region
     */
    NaturalLanguages[NaturalLanguages["Dutch_Netherlands"] = 229] = "Dutch_Netherlands";
    /**
     * Dutch Language for Suriname Region
     */
    NaturalLanguages[NaturalLanguages["Dutch_Suriname"] = 230] = "Dutch_Suriname";
    //#endregion
    //#region Portuguese
    /**
     * Portuguese Language for Angola Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_Angola"] = 231] = "Portuguese_Angola";
    /**
     * Portuguese Language for Brazil Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_Brazil"] = 232] = "Portuguese_Brazil";
    /**
     * Portuguese Language for Cape Verde Islands Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_CapeVerdeIslands"] = 233] = "Portuguese_CapeVerdeIslands";
    /**
     * Portuguese Language for Guinea-Bissau Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_GuineaBissau"] = 234] = "Portuguese_GuineaBissau";
    /**
     * Portuguese Language for Mozambique Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_Mozambique"] = 235] = "Portuguese_Mozambique";
    /**
     * Portuguese Language for Portugal Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_Portugal"] = 236] = "Portuguese_Portugal";
    /**
     * Portuguese Language for São Tomé e Principe Region
     */
    NaturalLanguages[NaturalLanguages["Portuguese_SaoTomeEPrincipe"] = 237] = "Portuguese_SaoTomeEPrincipe";
    //#endregion
    //#region Sindhi
    /**
     * Sindhi Language for India Region
     */
    NaturalLanguages[NaturalLanguages["Sindhi_India"] = 238] = "Sindhi_India";
    /**
     * Sindhi Language for Pakistan Region
     */
    NaturalLanguages[NaturalLanguages["Sindhi_Pakistan"] = 239] = "Sindhi_Pakistan";
    //#endregion
    //#region Serbian
    /**
     * Serbian Language for Bosnia-Herzegovina Region
     */
    NaturalLanguages[NaturalLanguages["Serbian_BosniaHerzegovina"] = 240] = "Serbian_BosniaHerzegovina";
    /**
     * Serbian Language for Yugoslavia Region
     */
    NaturalLanguages[NaturalLanguages["Serbian_Yugoslavia"] = 241] = "Serbian_Yugoslavia";
    //#endregion
    //#region Swati
    /**
     * Swati Language for Swaziland Region
     */
    NaturalLanguages[NaturalLanguages["Swati_Swaziland"] = 242] = "Swati_Swaziland";
    /**
     * Swati Language for South Africa Region
     */
    NaturalLanguages[NaturalLanguages["Swati_SouthAfrica"] = 243] = "Swati_SouthAfrica";
    //#endregion
    //#region Swedish
    /**
     * Swedish Language for Finland Region
     */
    NaturalLanguages[NaturalLanguages["Swedish_Finland"] = 244] = "Swedish_Finland";
    /**
     * Swedish Language for Sweden Region
     */
    NaturalLanguages[NaturalLanguages["Swedish_Sweden"] = 245] = "Swedish_Sweden";
    //#endregion
    //#region Swahili
    /**
     * Swahili Language for Kenya Region
     */
    NaturalLanguages[NaturalLanguages["Swahili_Kenya"] = 246] = "Swahili_Kenya";
    /**
     * Swahili Language for Tanzania Region
     */
    NaturalLanguages[NaturalLanguages["Swahili_Tanzania"] = 247] = "Swahili_Tanzania";
    //#endregion
    //#region Tamil
    /**
     * Tamil Language for India Region
     */
    NaturalLanguages[NaturalLanguages["Tamil_India"] = 248] = "Tamil_India";
    /**
     * Tamil Language for Singapore Region
     */
    NaturalLanguages[NaturalLanguages["Tamil_Singapore"] = 249] = "Tamil_Singapore";
    //#endregion
    //#region Tswana
    /**
     * Tswana Language for Botswana Region
     */
    NaturalLanguages[NaturalLanguages["Tswana_Botswana"] = 250] = "Tswana_Botswana";
    /**
     * Tswana Language for South Africa Region
     */
    NaturalLanguages[NaturalLanguages["Tswana_SouthAfrica"] = 251] = "Tswana_SouthAfrica";
    //#endregion
    //#region Turkish
    /**
     * Turkish Language for Bulgaria Region
     */
    NaturalLanguages[NaturalLanguages["Turkish_Bulgaria"] = 252] = "Turkish_Bulgaria";
    /**
     * Turkish Language for Cyprus Region
     */
    NaturalLanguages[NaturalLanguages["Turkish_Cyprus"] = 253] = "Turkish_Cyprus";
    /**
     * Turkish Language for Turkey Region
     */
    NaturalLanguages[NaturalLanguages["Turkish_Turkey"] = 254] = "Turkish_Turkey";
    //#endregion
    //#region Urdu
    /**
     * Urdu Language for India Region
     */
    NaturalLanguages[NaturalLanguages["Urdu_India"] = 255] = "Urdu_India";
    /**
     * Urdu Language for Pakistan Region
     */
    NaturalLanguages[NaturalLanguages["Urdu_Pakistan"] = 256] = "Urdu_Pakistan";
    //#endregion
    //#region Mandarin Chinese
    /**
     * Mandarin Chinese Language for China Region
     */
    NaturalLanguages[NaturalLanguages["MandarinChinese_China"] = 257] = "MandarinChinese_China";
    /**
     * Mandarin Chinese Language for Singapore Region
     */
    NaturalLanguages[NaturalLanguages["MandarinChinese_Singapore"] = 258] = "MandarinChinese_Singapore";
    /**
     * Mandarin Chinese Language for Taiwan Region
     */
    NaturalLanguages[NaturalLanguages["MandarinChinese_Taiwan"] = 259] = "MandarinChinese_Taiwan";
    //#endregion
    //#endregion
})(NaturalLanguages = exports.NaturalLanguages || (exports.NaturalLanguages = {}));
